virttest/qemu_devices/qcontainer.py

```diff
@@ -2926,42 +2926,48 @@ def define_hbas(
             )
             for key, value in blk_extra_params:
                 devices[-1].set_param(key, value)
-        if self.is_dev_iothread_vq_supported(devices[-1]):
-            if num_queues:
-                devices[-1].set_param("num-queues", num_queues)
-            # add iothread-vq-mapping if available
-            if image_iothread_vq_mapping:
-                val = []
-                for item in image_iothread_vq_mapping.strip().split(" "):
-                    allocated_iothread = self.allocate_iothread_vq(
-                        item.split(":")[0], devices[-1]
-                    )
-                    mapping = {"iothread": allocated_iothread.get_qid()}
-                    if len(item.split(":")) == 2:
-                        vqs = [int(_) for _ in item.split(":")[-1].split(",")]
-                        mapping["vqs"] = vqs
-                    val.append(mapping)
-                # FIXME: The reason using set_param() is that the format(
-                #  Example: iothread0:0,1,2 ) can NOT be set by
-                #  Devcontainer.insert() appropriately since the contents
-                #  following after colon are lost.
-                if ":" in image_iothread_vq_mapping:
-                    devices[-1].set_param("iothread-vq-mapping", val)
 
-            if isinstance(
-                self.iothread_manager, vt_iothread.MultiPeerRoundRobinManager
-            ):
-                mapping = self.iothread_manager.pci_dev_iothread_vq_mapping
-                if devices[-1].get_qid() in mapping:
-                    num_iothread = len(mapping[devices[-1].get_qid()])
-                    for i in range(num_iothread):
-                        iothread = self.allocate_iothread_vq("auto", devices[-1])
-                        iothread.iothread_vq_bus.insert(devices[-1])
-            elif isinstance(self.iothread_manager, vt_iothread.FullManager):
-                iothreads = self.allocate_iothread_vq("auto", devices[-1])
-                if iothreads:
-                    for ioth in iothreads:
-                        ioth.iothread_vq_bus.insert(devices[-1])
+        # add iothread_vq_bus_iothread and allocate iothread vq
+        for dev in devices:
+            if self.is_dev_iothread_vq_supported(dev):
+                if num_queues:
+                    if dev.params.get("driver") in ("virtio-blk-pci",):
+                        dev.set_param("num-queues", num_queues)
+                    elif dev.params.get("driver") in ("virtio-scsi-pci",):
+                        dev.set_param("num_queues", num_queues)
```

Comment:
```suggestion
                    if dev.params.get("driver") in ("virtio-blk-pci",):
                        dev.set_param("num-queues", num_queues)
                    elif dev.params.get("driver") in ("virtio-scsi-pci",):
                        dev.set_param("num_queues", num_queues)
                    else:
                        driver = dev.params.get("driver")
                        raise DeviceError(f"Unsupported the num queues for the driver {driver}")
```
We should consider that if we support a new driver in the future, but we do not handle the 
`num queues` for this driver, and it will not raise any error for the user.
Based on the above consideration, I suggest covering the `else` situation here.
Pls let me know your opinions. Thanks.